tests/fixtures/database.py

```diff
@@ -0,0 +1,1048 @@
+    CustomList,
+    DataSource,
+    DeliveryMechanism,
+    Edition,
+    ExternalIntegration,
+    ExternalIntegrationLink,
+    Genre,
+    Hyperlink,
+    Identifier,
+    IntegrationClient,
+    Library,
+    LicensePool,
+    MediaTypes,
+    Patron,
+    Representation,
+    RightsStatus,
+    SessionManager,
+    Subject,
+    Work,
+    WorkCoverageRecord,
+    create,
+    get_one_or_create,
+)
+from core.model.devicetokens import DeviceToken
+from core.model.licensing import License, LicensePoolDeliveryMechanism, LicenseStatus
+from core.util.datetime_helpers import utc_now
+
+
+class ApplicationFixture:
+    """The ApplicationFixture is a representation of the state that must be set up in order to run the application for
+    testing."""
+
+    @staticmethod
+    def create():
+        # This will make sure we always connect to the test database.
+        os.environ["TESTING"] = "true"
+
+        # Ensure that the log configuration starts in a known state.
+        LogConfiguration.initialize(None, testing=True)
+
+        # Drop any existing schema. It will be recreated when
+        # SessionManager.initialize() runs.
+        engine = SessionManager.engine()
+        # Trying to drop all tables without reflecting first causes an issue
+        # since SQLAlchemy does not know the order of cascades
+        # Adding .reflect is throwing an error locally because tables are imported
+        # later and hence being defined twice
+        # Deleting the problematic table first fixes the issue, in this case DeviceToken
+        DeviceToken.__table__.drop(engine, checkfirst=True)
+        Base.metadata.drop_all(engine)
+        return ApplicationFixture()
+
+    def close(self):
+        if "TESTING" in os.environ:
+            del os.environ["TESTING"]
+
+
+class DatabaseFixture:
+    """The DatabaseFixture stores a reference to the database."""
+
+    _engine: Engine
+    _connection: Connection
+
+    def __init__(self, engine: Engine, connection: Connection):
+        self._engine = engine
+        self._connection = connection
+
+    @staticmethod
+    def _get_database_connection() -> Tuple[Engine, Connection]:
+        url = Configuration.database_url()
+        engine, connection = SessionManager.initialize(url)
+        return engine, connection
+
+    @staticmethod
+    def create() -> "DatabaseFixture":
+        from core.model.customlist import customlist_sharedlibrary
+
+        # The CM uses SQLAlchemy's "declarative base" feature, which means that
+        # tables are only created if the classes containing references to them
+        # are actually loaded. The entire application database is therefore dependent on
+        # unspecified class initialization behaviour. The only safe way to guarantee
+        # that all tables are created is to refer to all of those classes before
+        # starting SQLAlchemy in order to ensure that all the classes are loaded
+        # in the interpreter.
+
+        tables = [
```

Comment:
Would it be more future-proof to loop through the models directory and import all files? This way any new models are automatically picked up.
A special `import Lane` will have to be done too, unfortunately.

Reply:
I agree. In this case I did the dumbest and most explicit thing. I think the database initialization should be refactored so that neither the app or the test suite are dependent on class initialization. I made a quick attempt at doing this, but it quickly introduced circular dependencies, so I backed it out.

Reply:
Yea, I tried something similar a while ago, self referential loops are all over the place, and an interpreted language is always going to rely on import order, don't think we can get around that.

Reply:
I agree it might be nice to future proof this a bit with a loop to import the models, assuming it doesn't add too much complexity. 

@io7m I remember us discussing centralizing the database initialization, so we don't have to do it separately in tests vs prod code, and that causing a cascade of import loops. I forget though if we made a tech debt ticket to handle that refactoring or not. Do you remember? If not, can you put one into the queue, so that we come back around to it?

Reply:
There is a ticket somewhere... :hourglass:

Reply:
Replaced this with something that iterates over the modules with `importlib`.